Fix Missing Width And Height Attributes On Images - Frontend [2/2]

engahmeds3ed opened the pull request:
Fixes #3307

crystinutzaa reviewed (changes requested):
This seems fine, however I think it still needs to handle the integration with `LayerSlider`
- Revisit the compatibility with  LayerSlider - https://github.com/wp-media/wp-rocket/blob/3b8b2d7debc83641045f9af82baa08b14e41daed/inc/3rd-party/plugins/slider/layerslider.php#L10-L13

engahmeds3ed commented:
Thanks @crystinutzaa 
@wp-media/php  I adjusted the code and changed the layerslider code a bit please check.
Also I added integration test case that uses the fixture related to the main image dimensions feature to assure that all cases not changed (input images = output images) without applying the feature of image dimensions even if it was active.
I don't know if we need it or not but I added it so feel free to advise about that.

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,306 @@
+		}
+
+		$replaces = [];
+		$images   = apply_filters( 'rocket_specify_dimension_images', $images_match[0] );
```
Missing docblock for the filter

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,306 @@
+	 * @return bool Valid to be parsed or not.
+	 */
+	private function can_specify_dimensions_external_images() {
+		return ini_get( 'allow_url_fopen' ) && apply_filters( 'rocket_specify_image_dimensions_for_distant', true );
```
Missing docblock for filter

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,306 @@
+		 *
+		 * @param bool Do the job or not.
+		 */
+		if ( ! apply_filters( 'rocket_specify_image_dimensions', true ) ) {
+			return false;
+		}
+
+		return true;
```
```suggestion
		return apply_filters( 'rocket_specify_image_dimensions', true );
```

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,306 @@
+				return false;
+			}
+
+			return getimagesize( $image_url );
```
why not return directly the dimensions index instead of returning the whole array from `getimagesize()`?

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,306 @@
+			||
+			false !== strpos( $image, 'data-no-image-dimensions' )
+			||
+			! preg_match( '/\s+src\s*=\s*[\'"]([^\'"]+)/i', $image, $src_match )
```
```suggestion
			! preg_match( '/\s+src\s*=\s*[\'"](?<url>[^\'"]+)/i', $image, $src_match )
```

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,306 @@
+			return false;
+		}
+
+		return $src_match[1];
```
```suggestion
		return $src_match['url'];
```

Named capturing groups help to make the code more readable

Tabrisrp commented on inc/ThirdParty/Plugins/Slider/LayerSlider.php:
```diff
@@ -0,0 +1,34 @@
+<?php
```
Did you test if we still need this compatibiliy file? Seeing how old the original code is, a lot can have happened since

engahmeds3ed replied:
@Tabrisrp Yes I tried that locally and I can see that layerslider alreay uses img tag with dimensions already set as in the following screenshot.

![image](https://user-images.githubusercontent.com/15707971/100135427-499c1200-2e92-11eb-90ea-4bcce031c537.png)

So our code won't affect it at all, even our code affected it and added dimensions, this won't make a conflict with layerslider.

@wp-media/php what do u think?

Tabrisrp replied:
Ok, so we don't need the compatibility file, right?

engahmeds3ed replied:
@Tabrisrp Yes I think so.

Tabrisrp replied:
great, so let's just remove it along the old compatibility file 👍🏼

engahmeds3ed replied:
Ok will so that now.

Tabrisrp commented on inc/Engine/Media/Images/Frontend.php:
```diff
@@ -0,0 +1,328 @@
+	 *
+	 * @param string $image_url Image url to get sizes for.
+	 *
+	 * @return array|false Get image sizes otherwise false.
```
```suggestion
	 * @return string|false Get image sizes otherwise false.
```

With the change, it's a string instead of an array, right?

engahmeds3ed replied:
Yes exactly, Done
